sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -286,7 +295,11 @@ private HttpRequest configRequest(final HttpRequest request, final SwaggerMethod
                 }
             }
 
-            if (isJson) {
+            if (bodyContentObject instanceof BinaryData) {
```

Comment:
Once it is merge this should use `RequestContent`. Right now, this is creating another copy of the underlying `BinaryData` data.